src/WebJobs.Script/Host/ScriptHostManager.cs

```diff
@@ -182,7 +184,7 @@ public void RunAndBlock(CancellationToken cancellationToken = default(Cancellati
                     // it will finish any currently executing functions and then clean itself up.
                     // Spin around and create a new host instance.
 #pragma warning disable 4014
-                    Orphan(newInstance);
+                    Task.Run(() => Orphan(newInstance));
```

Comment:
This is the most important change. Previously we were calling this async method from our sync RunAndBlock loop. If you look at the logic in Orphan, you'll see that this meant that we'd call instance.StopAsync on the host instance without waiting for that to finish, and we'd then immediately call dispose. So I spin this up on a background thread so the async function can run properly

Reply:
This Orphan call is in a try/catch. By moving it inside a Task.Run, aren't any exceptions that occur during the orphan step going to be unhandled?

Reply:
Technically, the `Orphan` call was already returning an unobserved `Task`, and since that was not awaited, exposure to this issue was already there. Ideally we'd address that and at least setup a continuation here similar to what we have in the catch block.

Reply:
Applying same continuation as in the other Orphan call